clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1052,6 +989,55 @@ void Sema::MarkDevice(void) {
     }
   }
 }
+//
+// Do we know that we will eventually codegen the given function?
+static bool isKnownEmitted(Sema &S, FunctionDecl *FD) {
+  if (!FD)
+    return true; // Seen in LIT testing
+
+  if (FD->hasAttr<SYCLDeviceAttr>() ||
+      FD->hasAttr<SYCLKernelAttr>())
+     return true;
+
+  // Templates are emitted when they're instantiated.
+  if (FD->isDependentContext())
+    return false;
+
+  // Otherwise, the function is known-emitted if it's in our set of
+  // known-emitted functions.
+  return S.DeviceKnownEmittedFns.count(FD) > 0;
+}
+
+Sema::DeviceDiagBuilder Sema::SYCLDiagIfDeviceCode(SourceLocation Loc,
```

Comment:
Hi Premanand, I need something like this function to avoid emitting an error for a function that has varargs and which is not part of DEVICE code.
In my case I have "include <stdio.h>" in the beginning of the source file and no uses of any functions like printf(f, ...) in DEVICE code, but clang still reports an error on Windows for such code.

Q1: Would your fix help with this error: "variadic function cannot use spir_function calling convention" emitted from SemaType.cpp:7014 ?
If NO, can this patch be extended to handle that case?

Q2: This MergeRequest was created almost 2 weeks ago. Is there some ETA for this fix approval/merge?